src/index.js

```diff
@@ -1,4 +1,151 @@
+      <div class="todo" id=${index}>
+        <i 
+          class="bi ${todo.checked ? 'bi bi-check2-square' : 'bi bi-app'}"
+          style="color : ${todo.color}"
+          data-action="check"
+        ></i>
+        <p class="${todo.checked ? 'checked' : ''}" data-action="check">${todo.value}</p>
+        <i class="bi bi-pencil-square" data-action="edit"></i>
+        <i class="bi bi-trash" data-action="delete"></i>
+      </div>
+      `;
+  });
+};
+
+// CHECK A TODO
+const checkTodo = (todoId) => {
+  todos = todos.map((todo, index) => ({
+    ...todo,
+    checked: index === todoId ? !todo.checked : todo.checked,
+  }));
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+// SAVE TODO
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+  // check if the todo is empty
+  const isEmpty = todoValue === '';
+
+  // check for duplicate todos
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+
+  if (isEmpty) {
+    showNotification("Todo's input is empty");
+  } else if (isDuplicate) {
+    showNotification('Todo already exists!');
+  } else {
+    if (EditTodoId >= 0) {
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        value: index === EditTodoId ? todoValue : todo.value,
+      }));
+      EditTodoId = -1;
+    } else {
+      todos.push(
+        {
+          value: todoValue,
+          checked: false,
+          color: 'orange',
+        },
+      );
+    }
+
+    todoInput.value = '';
+  }
+};
+
+// FORM SUBMIT
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  saveTodo();
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+});
+
+// DELETE TODO
+const deleteTodo = (todoId) => {
+  todos = todos.filter((todo, index) => index !== todoId);
+  EditTodoId = -1;
+  // re-render
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+// CLICK EVENT LISTENER FOR ALL THE TODOS
+todosListEl.addEventListener('click', (event) => {
+  const { target } = event;
+  const { parentElement } = target;
+  if (parentElement.className !== 'todo') return;
+  // t o d o id
+  const todo = parentElement;
+  const todoId = Number(todo.id);
+  // target action
+  const { action } = target.dataset;
+  if (action === 'check') {
+    checkTodo(todoId);
+  }
+  if (action === 'edit') {
+    editTodo(todoId);
+  }
+  if (action === 'delete') {
+    deleteTodo(todoId);
+  }
+});
+
+// const clearAllBtn = document.querySelector('.clear-completed');
+// const clearAll = () => {
+//     const localData = JSON.parse(localStorage.getItem('todos'));
+//     todosListEl.forEach(i => {
+//         if(i.classList.contains('check')){
+//             deleteTodo(i)
+//         }
+//     });
+//     let count = 0;
+//     const data = todos.from(localData).filter(i => !i.completed);
+//     data.map(i => i.index = count += 1);
+//     localStorage.setItem('todos', JSON.stringify(data));
+// }
```

Comment:
- [ ] Please make sure to delete the comments lines of code in your file to keep a clean codebase 🙃🙃